spec/services/user_invite_service_spec.rb

```diff
@@ -50,5 +50,60 @@
       expect(result).to have_role(Role::ORG_ADMIN, organization)
       expect(result).not_to have_role(Role::PARTNER, :any)
     end
+
+    it "should create the user without name with role" do
+      result = nil
+      expect {
+        result = described_class.invite(name: "",
+          email: "email2@email.com",
+          roles: [Role::ORG_USER, Role::ORG_ADMIN],
+          resource: organization)
+      }.to change { ActionMailer::Base.deliveries.count }.by(1)
+      expect(result.name).to eq("")
+      expect(result.email).to eq("email2@email.com")
+      expect(result).to have_role(Role::ORG_USER, organization)
+      expect(result).to have_role(Role::ORG_ADMIN, organization)
+    end
+  end
+end
+
+RSpec.describe UserInviteService, type: :service do
+  let!(:organization) { FactoryBot.create(:organization) }
+  let!(:partner) { FactoryBot.create(:partner, organization: organization) }
+
+  before do
+    allow(UserMailer).to receive(:role_added).and_return(double(:mail, deliver_later: nil))
+  end
+
+  context "with a new user" do
+    it "should invite a user with the partner role" do
+      expect {
+        described_class.invite(
+          name: "Partner User",
+          email: "partner@example.com",
+          roles: [Role::PARTNER],
+          resource: partner
+        )
+      }.to change(User, :count).by(1).and change(ActionMailer::Base.deliveries, :count).by(1)
+
+      new_user = User.find_by(email: "partner@example.com")
+      expect(new_user).not_to be_nil
+      expect(new_user.has_role?(:partner, partner)).to be true
+    end
+
+    it "should create the user without a name with default role" do
+      expect {
+        described_class.invite(
+          name: "",
+          email: "email2@example.com",
+          roles: [Role::ORG_USER],
+          resource: organization
+        )
+      }.to change(User, :count).by(1)
+
+      new_user = User.find_by(email: "email2@example.com")
+      expect(new_user.name).to eq("")
```

Comment:
we are discussing on a call right now that this expectation should be

```suggestion
      expect(new_user.name).to eq("Name Not Provided")
```